tensornetwork/backend_contextmanager.py

```diff
@@ -0,0 +1,32 @@
+from typing import Text, Union
+from tensornetwork.backends.base_backend import BaseBackend
+import tensornetwork.config as config
```

Comment:
Can we move the `set_default_backend` and everything in `config` to this file? No need to make these separate anymore and I like this filename more.